test/unit/controllers/todo-item.controller.test.ts

```diff
@@ -0,0 +1,260 @@
+      const findTodoItem = todoItemRepo.findById as sinon.SinonStub;
+      const item = Object.assign({}, payload, { id: 1, todoId: 1 });
+      const newItem = Object.assign({}, payload, { title: 'title two' });
+      update.withArgs(item.id as number, newItem).rejects();
+      findTodoItem.withArgs(item.id as number).resolves(null);
+      try {
+        await controller.updateTodoItem(
+          item.id as number,
+          newItem,
+          todo.id as number
+        );
+      } catch (err) {
+        expect(err).to.match(/No todo item was found with id/);
+      }
+    });
+  });
+  describe('deleteTodoItem', () => {
+    it('deletes a todo item if it exists', async () => {
+      const controller = new TodoItemController(todoItemRepo, todoRepo);
+      const del = todoItemRepo.deleteById as sinon.SinonStub;
+      const findTodoItem = todoItemRepo.findById as sinon.SinonStub;
+      const item = Object.assign({}, payload, { id: 1, todoId: 1 });
+      del.withArgs(item.id as number).resolves(true);
+      findTodoItem.withArgs(item.id as number).resolves(todo);
+      expect(
+        await controller.deleteTodoItem(item.id as number, todo.id as number)
+      ).to.eql(true);
+    });
+
+    it('throws if it does not exist', async () => {
+      const controller = new TodoItemController(todoItemRepo, todoRepo);
+      const del = todoItemRepo.deleteById as sinon.SinonStub;
+      const findTodoItem = todoItemRepo.findById as sinon.SinonStub;
+      const item = Object.assign({}, payload, { id: 1, todoId: 1 });
+      const newItem = Object.assign({}, payload, { title: 'title two' });
+      del.withArgs(item.id as number, newItem).rejects();
+      findTodoItem.withArgs(item.id as number).resolves(null);
+      try {
+        await controller.deleteTodoItem(item.id as number, todo.id as number);
+      } catch (err) {
+        expect(err).to.match(/No todo item was found with id/);
+      }
+    });
+  });
+  describe('deleteAllTodoItems', () => {
+    it('deletes all existing todo items', async () => {
+      const controller = new TodoItemController(todoItemRepo, todoRepo);
+      const del = todoItemRepo.deleteAll as sinon.SinonStub;
+      const find = todoRepo.findById as sinon.SinonStub;
+      del.resolves(true);
+      find.withArgs(todo.id as number).resolves(todo);
+      expect(await controller.deleteAllTodoItems(todo.id as number)).to.eql(
+        true
+      );
+    });
+
+    it('throws if no todo exists to delete all items from', async () => {
+      const controller = new TodoItemController(todoItemRepo, todoRepo);
+      const del = todoItemRepo.deleteAll as sinon.SinonStub;
+      const find = todoRepo.findById as sinon.SinonStub;
+      del.resolves(false);
+      find.withArgs(todo.id as number).resolves(null);
+      try {
+        await controller.deleteAllTodoItems(todo.id as number);
+      } catch (err) {
+        expect(err).to.match(/No todo was found with id/);
+      }
+    });
+  });
+});
+/*
```

Comment:
We don't keep commented-out blocks of code. Please uncomment and make the code work, or remove it completely.